lib/apis/v3/kubecontrollersconfig.go

```diff
@@ -0,0 +1,165 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v3
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	KindKubeControllersConfiguration     = "KubeControllersConfiguration"
+	KindKubeControllersConfigurationList = "KubeControllersConfigurationList"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// KubeControllersConfiguration contains the configuration for Calico Kubernetes Controllers.
+type KubeControllersConfiguration struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Specification of the KubeControllersConfiguration.
+	Spec KubeControllersConfigurationSpec `json:"spec,omitempty"`
+	// Status of the KubeControllersConfiguration
+	Status KubeControllersConfigurationStatus `json:"status,omitempty"`
+}
+
+// KubeControllersConfigurationSpec contains the values of the Kubernetes controllers configuration.
+type KubeControllersConfigurationSpec struct {
+
+	// LogSeverityScreen is the log severity above which logs are sent to the stdout. [Default: Info]
+	LogSeverityScreen string `json:"logSeverityScreen,omitempty" validate:"omitempty,logLevel"`
+
+	// HealthChecks enables or disables support for health checks
+	HealthChecks string `json:"healthChecks,omitempty" validate:"omitempty,oneof=Enabled Disabled"`
+
+	// EtcdV3CompactionPeriod is the period between etcdv3 compaction requests. Set to 0 to disable.
+	EtcdV3CompactionPeriod string `json:"etcdV3CompactionPeriod,omitempty" validate:"omitempty,duration"`
+
+	// Controllers enables and configures individual Kubernetes controllers
+	Controllers ControllersConfig `json:"controllers"`
+}
+
+// ControllersConfig enables and configures individual Kubernetes controllers
+type ControllersConfig struct {
+	// Node enables and configures the node controller
+	Node *NodeControllerConfig `json:"node,omitempty"`
+
+	// Policy enables and configures the policy controller
+	Policy *PolicyControllerConfig `json:"policy,omitempty"`
+
+	// WorkloadEndpoint enables and configures the workload endpoint controller
+	WorkloadEndpoint *WorkloadEndpointControllerConfig `json:"workloadEndpoint,omitempty"`
+
+	// ServiceAccount enables and configures the service account controller
+	ServiceAccount *ServiceAccountControllerConfig `json:"serviceAccount,omitempty"`
+
+	// Namespace enables and configures the namespace controller
+	Namespace *NamespaceControllerConfig `json:"namespace,omitempty"`
+}
+
+// NodeControllerConfig configures the node controller, which automatically cleans up configuration
+// for nodes that no longer exist. Optionally, it can create host endpoints for all Kubernetes nodes.
+type NodeControllerConfig struct {
+
```

Comment:
nit: extra lines here and at the top of other structs

Reply:
Will fix :+1: